1/numberDivisors.js

```diff
@@ -0,0 +1,24 @@
+// complexity is O(log n) - I'm not sure
+function numberDivisors(number) {
+  const result = [];
+  for (let divisor = 1; divisor < (number/divisor + 1); divisor++) {
+    if (number % divisor === 0 && !result[divisor]) { result.push(divisor); result.push(number/divisor);}
```

Comment:
What does `!result[divisor]` condition check? Why do we need it here?

Reply:
~~~example:
the divisor is `1`, `number/divisor` is `X`. I put in the `result` 1 and X. When we face `X` as a divisor we shouldn't add it again~~~  
got it, where will not be a variant like `divisor = X`